src/java/org/apache/cassandra/config/EncryptionOptions.java

```diff
@@ -76,8 +92,16 @@ public String description()
     protected Boolean isEnabled = null;
     protected Boolean isOptional = null;
 
+    /*
+     * We will wait to initialize this until applyConfig() call to make sure we do it only when the caller is ready
+     * to use this option instance.
+     */
+    public ISslContextFactory sslContextFactoryInstance;
+
     public EncryptionOptions()
     {
+        ssl_context_factory = new ParameterizedClass("org.apache.cassandra.security.DefaultSslContextFactoryImpl",
```

Comment:
There is an empty constructor in DefaultSsl.... so after its creation, `parameters` will be null but here you are setting it like an empty map, even it is a detail, it should be same and that empty constructor should call the one with map arg and create new map there.